spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/actions/TestRewriteDataFilesAction.java

```diff
@@ -195,6 +124,33 @@ public void testBinPackWithFilter() {
     assertEquals("Rows must match", expectedRecords, actualRecords);
   }
 
+  // A version of this using more complicated partitioning is in the Spark Extensions Module
+  @Test
+  public void testBinPackAfterPartitionChange() {
+    Table table = createTable();
+    writeRecords(20, SCALE, 4);
+    shouldHaveFiles(table, 20);
+    table.updateSpec().addField(Expressions.ref("c1")).commit();
+
+    List<Object[]> originalData = currentData();
+
+    RewriteDataFiles.Result result =
+        basicRewrite(table)
+            .option(SortStrategy.MIN_INPUT_FILES, "1")
+            .option(SortStrategy.MIN_FILE_SIZE_BYTES, Integer.toString(averageFileSize(table) + 1000))
+            .execute();
+
+    Assert.assertEquals("Should have 1 fileGroup because all files were not correctly partitioned",
```

Comment:
Same as above (actual, expected reversed).  And also "Assert" unnecessary.

Reply:
We are using the Assert assertEquals and not the row comparison SparkTestBase.assertEquals here